### PR TITLE
Environments: add run deps to shell modifications

### LINKED ISSUE #23368: `spack env activate` does not traverse run dependencies for environment modifications
Moving this from #gripes on Slack.

Mine @SteVwonder question was:

> Why does spack env activate not run setup_run_environment for dependencies of packages listed in an env yaml?  Due to this behavior spack load flux-sched behaves differently from spack env activate . with a yaml file containing flux-sched as the only package.

@tgamblin answer was:

> `_env_modifications_for_default_view` doesn’t currently traverse run deps, but it really should

### PR DESCRIPTION
**Pull request by scheibelp**
Fixes #23368

When adding an Environment to a user's shell, Spack was only adding root specs. This now includes run dependencies of root specs.

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]
It would be good to have a test for this, but otherwise LGTM

**Comment by scheibelp** [comment]
@alalazo does the new test look reasonable?

File: lib/spack/spack/environment.py
Comment (alalazo) [inline lib/spack/spack/environment.py]: I wonder if we should warn or not if entering this branch. I can see arguments for both behaviors. so not sure what to do. I am fine deferring this choice to a later PR, leaving this comment in case you have a clear opinion on that.

**Review by alalazo — DISMISSED** [review]
@scheibelp Test looks good to me, and I am fine merging the PR as is. I just left a minor comment, so I'll wait for your opinion on that before merging.

File: lib/spack/spack/environment.py
Comment (scheibelp) [reply lib/spack/spack/environment.py]: I think a warning message might be too verbose but a debug message would be good. Let me add that today

**Review by alalazo — APPROVED** [review]

**Comment by tgamblin** [comment]
@vchuravy FYI

**Pull request merged**
